src/StripeCheckoutSessionGatewayFactory.php

```diff
@@ -40,12 +39,15 @@ protected function populateConfig(ArrayObject $config): void
             'payum.action.authorize' => new AuthorizeAction(),
             'payum.action.convert_payment' => new ConvertPaymentAction(),
             'payum.action.redirect_to_checkout' => new RedirectToCheckoutAction(),
+            'payum.action.cancel' => new CancelAction(),
```

Comment:
@dragosprotung the original `CancelAction` must not be replaced because it handle authorize flow within both gateways StripeCheckoutSession and StripeJS.
One way to fix this is to add a check for the `manual` `capture_method` inside the original one (`support` method) and rename your new action ID to `payum.action.cancel_checkout_session`.

Let me know if you have a better idea.

Reply:
But the `CancelAction` is only replaced for the `CheckoutSession`. I added a test for that.

Reply:
Yes but the old one is no more available because you are using the same action ID making it replaced in the array option and therefore never used by Payum.

Reply:
So you mean that for the `CheckoutSession` we should still keep the old one? 
Then I would apply a check for the `manual` `capture_method` like you suggested in the old one. 
I would also rename the config value for both something like: `payum.action.cancel.manual` and `payum.action.cancel.automatic` (please suggest other names maybe).

Reply:
Exactly you understand it perfectly, the naming is explaining what the cancel is doing, so maybe the old one should be renamed `payum.action.cancel.payment_intent.manual` and the new one you created `payum action.cancel.payment_intent.automatic` because I will need to build the cancel action for  the subscription in an other PR

Reply:
Done. Hope all scenarios are covered for now for this PR

Reply:
Yes they are now thank you very much 😉